pkg/cmd/cli/cmd/set/volume.go

```diff
@@ -237,10 +237,6 @@ func (v *VolumeOptions) Validate(args []string) error {
 
 func (a *AddVolumeOptions) Validate(isAddOp bool) error {
 	if isAddOp {
-		if len(a.MountPath) == 0 {
```

Comment:
I feel like its also a valid use case to add a volume sans overwrite and not provide the mount-path.  I'd agree that it's not a very good use case since a user would still need to define a volumeMount with a mount-path to make use of it, but it would still be able to create a volume definition for them.  To me providing a mount-path tells set volume to do two separate tasks.